src/test/java/net/snowflake/ingest/streaming/internal/RegisterServiceTest.java

```diff
@@ -59,7 +58,17 @@ public void teardown() throws Exception {
 
   @Test
   public void testRegisterService() throws ExecutionException, InterruptedException {
-    RegisterService<StubChunkData> rs = new RegisterService<>(null, true);
+    SnowflakeStreamingIngestClientInternal<StubChunkData> client =
+        Mockito.mock(SnowflakeStreamingIngestClientInternal.class);
+    ParameterProvider parameterProvider = Mockito.mock(ParameterProvider.class);
```

Comment:
curious why do we need to mock the parameter provider, can we not do away with this unnecessary mocking (not for this PR) ?

Reply:
We need this as `RegisterServiceTest` is reading blob upload timeout from parameter provider.